pyslackers_website/slack/views.py

```diff
@@ -44,3 +49,32 @@ def post(self, request, *args, **kwargs):
             messages.success(request, 'Invite sent, see you in Slack!')
             return self.form_valid(form)
         return self.form_invalid(form)
+
+    def get_membership_history_json(self):
+        """ Build membership history JSON """
+        counts = []
+        xlabels = []
+        is_first_record = True
+
+        membership_history = Membership.objects.values_list('member_count', 'timestamp')
```

Comment:
I think we need to do two things here:

1. We can actually do this work in SQL (which would be much faster and require less logic)


In SQL (which I am not sure how to convert to the django ORM yet)

```sql
SELECT timestamp::date, member_count
FROM slack_membership
  WHERE timestamp::date IN (
    SELECT date_trunc('month', timestamp)::date AS month
    FROM slack_membership GROUP BY month
  )
ORDER BY timestamp;
```

which with real data gives us:

```
 timestamp  | member_count 
------------+--------------
 2017-08-01 |         7491
 2017-09-01 |         8350
 2017-10-01 |         9107
 2017-11-01 |         9712
 2017-12-01 |        10308
 2018-01-01 |        10842
```

2. Make this async (probably expose an api endpoint to get this data) - which I can do after this PR. I am planning to migrate the timezone stuff over too.

Secondly, the code should not manually convert this to JSON - that suggest the function knows how the result will be used (which really it shouldn't). Though I understand the reasoning for it

If we were going to pursue this way, I would suggest using the `| tojson` template tag we have in `core/templatetags/tojson.py` - but unfortunately that doesn't work for us since the JS is now separated from the templates more and goes through a build system (see the later comment).

Reply:
1) Agreed. I wasn't sure how to do it w/ an ORM, TBH.
2) "I am planning to migrate the timezone stuff over too" - yeah, saw you record as an issue a couple hrs ago. Retrieving directly from API would be nice.

Reply:
Just remembered something...
I wanted to pull dailies b/c (especially for first yr or 2), having a single number representing a whole month makes the graph jagged. With dailies, it looks a little more organic, but there's absolutely no need to place a date for each data pt.